test/integration/good/code-gen/cpp.expected

```diff
@@ -6506,65 +6535,8302 @@ class jacobian_pe_model final : public model_base_crtp<jacobian_pe_model> {
+      current_statement__ = 36;
+      stan::math::validate_non_negative_index("theta2", "n_obs", n_obs);
+    } catch (const std::exception& e) {
+      stan::lang::rethrow_located(e, locations_array__[current_statement__]);
+    }
+    num_params_r__ = 1 + 1 + 1;
+  }
+  inline std::string model_name() const final {
+    return "laplace_bernoulli_logit_model";
+  }
+  inline std::vector<std::string> model_compile_info() const noexcept {
+    return std::vector<std::string>{"stanc_version = %%NAME%%3 %%VERSION%%",
+             "stancflags = --print-cpp"};
+  }
+  // Base log prob
+  template <bool propto__, bool jacobian__, typename VecR, typename VecI,
+            stan::require_vector_like_t<VecR>* = nullptr,
+            stan::require_vector_like_vt<std::is_integral, VecI>* = nullptr,
+            stan::require_not_st_var<VecR>* = nullptr>
+  inline stan::scalar_type_t<VecR>
+  log_prob_impl(VecR& params_r__, VecI& params_i__, std::ostream*
+                pstream__ = nullptr) const {
+    using T__ = stan::scalar_type_t<VecR>;
+    using local_scalar_t__ = T__;
+    T__ lp__(0.0);
+    stan::math::accumulator<T__> lp_accum__;
+    stan::io::deserializer<local_scalar_t__> in__(params_r__, params_i__);
+    int current_statement__ = 0;
+    // suppress unused var warning
+    (void) current_statement__;
+    local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
+    // suppress unused var warning
+    (void) DUMMY_VAR__;
+    static constexpr const char* function__ =
+      "laplace_bernoulli_logit_model_namespace::log_prob";
+    // suppress unused var warning
+    (void) function__;
+    try {
+      current_statement__ = 1;
+      auto alpha =
+        in__.template read_constrain_lb<local_scalar_t__, jacobian__>(0,
+          lp__);
+      current_statement__ = 2;
+      auto rho =
+        in__.template read_constrain_lb<local_scalar_t__, jacobian__>(0,
+          lp__);
+      current_statement__ = 3;
+      auto eta =
+        in__.template read_constrain_lb<local_scalar_t__, jacobian__>(0,
+          lp__);
+      {
+        current_statement__ = 6;
+        lp_accum__.add(stan::math::inv_gamma_lpdf<propto__>(rho,
+                         rho_location_prior, rho_scale_prior));
+        current_statement__ = 7;
+        lp_accum__.add(stan::math::inv_gamma_lpdf<propto__>(alpha,
+                         alpha_location_prior, alpha_scale_prior));
+        current_statement__ = 8;
+        lp_accum__.add(stan::math::normal_lpdf<propto__>(eta, 0, 1));
+        current_statement__ = 9;
+        lp_accum__.add(stan::math::laplace_marginal_bernoulli_logit_lpmf<
+                         propto__>(y, y, theta_0, K_function_functor__(),
+                         std::tuple<
+                           const std::vector<Eigen::Matrix<double,-1,1>>&,
+                           int, local_scalar_t__, local_scalar_t__>(x, n_obs,
+                           alpha, rho), pstream__));
```

Comment:
Can we have this generate to `std::forward_as_tuple`?

Reply:
Also I think `std::vector<Eigen::Matrix<double,-1,1>>&,` would fail if that type was an rvalue inside of the constructor

Reply:
(See 9622dea031e21ccb261f35a8e4cec0b639313f73 and 599774d7e64706519b3a113be8914b105b3e9e15 for the origins of the current code in the Tuples PR)